home/screens/SnacksListScreen/SnackList.tsx

```diff
@@ -80,22 +80,21 @@ function SnackList({ data, loadMoreAsync }: Props) {
     }
   };
 
-  const renderItem = React.useCallback(
-    ({ item: snack, index }: { item: CommonSnackDataFragment; index: number }) => {
-      return (
-        <SnacksListItem
-          key={index.toString()}
-          url={snack.fullName}
-          name={snack.name}
-          description={snack.description}
-          isDraft={snack.isDraft}
-          first={index === 0}
-          last={index === data.length - 1}
-        />
-      );
-    },
-    []
-  );
+  const renderItem = ({ item: snack, index }: { item: CommonSnackDataFragment; index: number }) => {
```

Comment:
is it necessary to define this function within render? it looks like we don't close over any values and so we should pull this out to a function outside of render, otherwise we re-render every list item whenever the list changes because the render item function changes

Reply:
It closes over `data` to check where the item is in the list. I think it might be more appropriate to bring back the `useCallback`, but pass in `data` so it fixes the stale `renderItem` callback problem

Reply:
ah i missed that check. can we possibly move whatever special styling we need to a footer component instead? https://reactnative.dev/docs/next/flatlist#listfootercomponent

###### <a href="https://getunblocked.com"><img src="https://avatars.githubusercontent.com/in/166219" width=20 align="center" /></a>&nbsp;&nbsp;Comment posted from [Unblocked](https://getunblocked.com/dashboard/team/5bf29d2c-23ae-4af8-8ccc-fe767cd2df79/thread/ec784d69-143d-401f-8853-09c9e9f9a92e?message=2912d7a8-c343-4cb3-988b-5f1223393b6e).

Reply:
It's possible, but I don't know if we should. The bottom styling is just a rounded border. If we add another element to the top/bottom of the list for that effect, when you press the first or last item, it won't show feedback on the footer/header components which feels wrong for this UI pattern.

Here's what the bottom of the list looks like for reference:

![IMG_5531](https://user-images.githubusercontent.com/12488826/180522228-69726878-44ef-4333-a83c-a99eff947936.PNG)

Reply:
Also it's a larger refactor that doesn't seem right for the scope of this bugfix pr

Reply:
agreed. i thought it might have been low hanging fruit but we can address it in a follow up later if needed